app/views/entries/_entry.html.erb

```diff
@@ -30,7 +30,7 @@
             <strong><%= Link.model_name.human.pluralize %>:</strong>
 
             <% entry.links.each do |link| %>
-              <%= link_to link_name(link), link.address %>
+              <%= link_to image_tag(link_icon(link), :class => "link-icon") + link_name(link), link.address %>
```

Comment:
A few things here. Firstly, rather than joining the output of `image_tag` and `link_name`, we can use the block method of `link_to`. I think it makes it a bit easier to follow the markup generated as a result.

```suggestion
              <%= link_to link.address do %>
                <%= image_tag link_icon(link), class: "link-icon" %>
                <%= link_name(link) %>
              <% end %>
```

Also, it'd be great if we could use SVG icons. There's a few reasons I'd prefer this:

- We can use them at any size
- We can recolour and restyle them as we see fit
- We can use `inline_svg_pack_tag` to embed the SVG code directly inline and save on extra file loads

If we can get them, they can go in `app/javascript/svg/` and we can use the [`icon_tag` helper](https://github.com/jdbann/exlibris/blob/main/app/helpers/icon_helper.rb).

Then finally, we have some classes to help with icons already setup. One for [`.with-icon`](https://github.com/jdbann/exlibris/blob/main/app/javascript/scss/compositions/_icon.scss) which helps present an icon nicely alongside text and one for [`.icon`](https://github.com/jdbann/exlibris/blob/main/app/javascript/scss/blocks/_icon.scss) which applies the right colours. So this could become:

```suggestion
              <%= link_to link.address, class: "with-icon" do %>
                <%= icon_tag :itch %>
                <%= link_name(link) %>
              <% end %>
```

Reply:
I've noticed some conflicts with the `.icon` classes but I can tidy that up later